config/initializers/protected_id.rb

```diff
@@ -0,0 +1,88 @@
+###
+# Copyright 2016 - 2019 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module ProtectedId
+  PROTECT_IDS = ENV['PROTECTED_IDS'].blank? ||  ENV['PROTECTED_IDS'] != 'false'
+
+  module Encoder
+    def encode(id)
+      return id.to_s unless PROTECT_IDS
+
+      day_stamp = Date.today.to_time.to_i / (60 * 60 * 24) # Seconds in a day
+      obfuscate(id, day_stamp)
+    end
+    module_function :encode
+
+    def encoded?(id)
+      false if id.blank?
+
+      id.to_s.match(/.*==$/)
+    end
+    module_function :encoded?
+
+    def decode(encoded)
+      return encoded unless encoded?(encoded)
```

Comment:
This should check for `encoded.is_a?(Enumerable)` and return a compacted map of each encoded item if it is.